pkg/cloud/services/awsnode/cni.go

```diff
@@ -63,33 +63,21 @@ func (s *Service) ReconcileCNI(ctx context.Context) error {
 		return ErrCNIMissing
 	}
 
-	envVars := s.scope.VpcCni().Env
-	if len(envVars) > 0 {
+	var needsUpdate bool
+	if len(s.scope.VpcCni().Env) > 0 {
```

Comment:
That's a good point. 🤔 If they remove all env variables from the setting it wouldn't do anything right now, since it would think none have been defined. But this is a bit finicky. Because how do I know that there was previously something defined in them and completely removed? I don't think we store that it has been completely cleared instead of never set?

Reply:
I guess the only way is to look at what is on the ds currently and then compare against whats coming from `s.scope.VpcCni().Env`

Reply:
But that has extra stuff that will not be in there and we don't know which one is set by default or was just defined by the user.

Reply:
So the incoming container Env will have extra variables that weren't set by the use but by the ds itself. How do we know which were defined by the user and which are there by default is the question then.

Reply:
So now that I thought about this a bit more it actually occurred to me that you can't just delete an environment property once you set a value to it. 🤔 Because it might need a default value and if remove it completely from the list of environment properties, I don't know how aws-node will behave. Maybe it will even error out if you don't update it back to its original value. They probably have some guards in place MAYBE... But I'm not sure. :)

Reply:
Thats a good point. I think the deduplication is enough